### PR TITLE
Added cottonwood tree boll to cottonwood tree and a recipe.

### PR DESCRIPTION
**Pull request by Nebnis**
#### Summary
Balance "Added cotton tree boll as harvest for cottonwood trees"

#### Purpose of change
Tree was missing materials, i also added the new item as part of the plant fiber recipe.

#### Describe the solution
Touching a tree drops 3~6 cottonwood tree boll

#### Describe alternatives you've considered
Nothing

#### Testing
Touched a bunch of trees and everything seems to be working along with the recipes.

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
Is this a thing you can actually do?

**Comment by Nebnis** [comment]
I have found this answer for it .

"Yarn can be spun from cottonwood fibers which can then be knit or crocheted into clothing. The issue with cottonwood fibers is that they are much shorter than regular cotton fibers, which can make spinning them more difficult. Combining cottonwood fibers with regular cotton and spinning them together into yarn can help facilitate creating yarns from them more easily. Yarn from cottonwood is typically only produced by hobbyists, who claim that it is strong and has a mild lustre."

**Review by NetSysFire — DISMISSED** [review]
I do not think you can repurpose cotton scraps for this. The cottonwood tree is named that way because the fibers look similar to cotton, not because they are cotton.
You should make a separate item for the fiber ball thing you harvest from the tree which can be similar to the already existing dogbane. Add the new item to the plant fiber recipe and you are good to go.

**Comment by Nebnis** [comment]
Would be okay with just adding plant fiber from the tree rather than making a new item? Or should i really?

**Comment by NetSysFire** [comment]
Just plucking plant fiber from the tree eliminates the need to further refine the item.

>they are much shorter than regular cotton fibers, which can make spinning them more difficult

However, seeing the difficulties with obtaining actual usable fiber from the tree it makes sense to require an additional step. So yes, please do add an item for this.

**Comment by Nebnis** [comment]
Should i make a different pr for the item or just do everything here?

**Comment by NetSysFire** [comment]
Do everything in here because all those changes work toward the same goal.

File: data/json/furniture_and_terrain/terrain-flora.json [hunk 1]
```diff
@@ -444,7 +444,7 @@
     "id": "t_tree_cottonwood_harvested",
     "name": "cottonwood tree",
     "looks_like": "t_tree",
-    "description": "A large tree belonging to the 'Populus' genus, commonly found throughout the New England region. All cotton has been harvested.",
+    "description": "A large tree belonging to the 'Populus' genus, commonly found throughout the New England region. All the cottonwood tree boll has been harvested.",
```
Comment (NetSysFire) [inline data/json/furniture_and_terrain/terrain-flora.json:447]:
```suggestion
    "description": "A large tree belonging to the 'Populus' genus, commonly found throughout the New England region. All the cottonwood tree boll have been harvested.",
```

**Comment by Nebnis** [comment]
Sorry about all the errors, I'm trying to learn how all this work

File: data/json/furniture_and_terrain/terrain-flora.json [hunk 2]
```diff
@@ -416,13 +416,42 @@
     "id": "t_tree_cottonwood",
     "name": "cottonwood tree",
     "looks_like": "t_tree",
-    "description": "A large tree belonging to the 'Populus' genus, commonly found throughout the New England region.",
+    "description": "A large tree belonging to the 'Populus' genus, commonly found throughout the New England region. You see some cottonwood tree boll that can be harvested.",
```
Comment (TheShadowFerret) [inline data/json/furniture_and_terrain/terrain-flora.json:419]:
```suggestion
    "description": "A large tree belonging to the 'Populus' genus, commonly found throughout the New England region. You see some cottonwood tree bolls that can be harvested.",
```

File: data/json/furniture_and_terrain/terrain-flora.json [hunk 3]
```diff
@@ -416,13 +416,42 @@
...
+        { "item": "leaves", "count": [ 50, 150 ] }
+        ]
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_tree_cottonwood_harvested",
+    "name": "cottonwood tree",
+    "looks_like": "t_tree",
+    "description": "A large tree belonging to the 'Populus' genus, commonly found throughout the New England region. All the cottonwood tree boll have been harvested.",
```
Comment (TheShadowFerret) [inline data/json/furniture_and_terrain/terrain-flora.json:447]:
```suggestion
    "description": "A large tree belonging to the 'Populus' genus, commonly found throughout the New England region. All the cottonwood tree bolls have been harvested.",
```

**Review by TheShadowFerret — COMMENTED** [review]
Plurals

**Review by TheShadowFerret — APPROVED** [review]

**Pull request merged**